frontend/packages/pipelines-plugin/src/components/shared/workspaces/__tests__/WorkspaceDefinitionList.spec.tsx

```diff
@@ -0,0 +1,47 @@
+import * as React from 'react';
+import { ShallowWrapper, shallow } from 'enzyme';
+import WorkspaceDefinitionList, { WorkspaceDefinitionListProps } from '../WorkspaceDefinitionList';
+import { pipelineTestData, PipelineExampleNames } from '../../../../test-data/pipeline-data';
+
+jest.mock('react-i18next', () => {
+  const reactI18next = require.requireActual('react-i18next');
+  return {
+    ...reactI18next,
+    useTranslation: () => ({ t: (key) => key }),
+  };
+});
+
+describe('WorkspaceDefinitionList', () => {
+  let WorkspaceDefinitionListWrapper: ShallowWrapper<WorkspaceDefinitionListProps>;
+  let WorkspaceDefinitionListWrapperProps: WorkspaceDefinitionListProps;
+
+  it('Should not render dd or dt heading if no workspaces are found', () => {
+    WorkspaceDefinitionListWrapperProps = {
+      workspaces: null,
+    };
+    WorkspaceDefinitionListWrapper = shallow(
+      <WorkspaceDefinitionList {...WorkspaceDefinitionListWrapperProps} />,
+    );
+    expect(WorkspaceDefinitionListWrapper.find('dd').exists()).toBe(false);
+    expect(WorkspaceDefinitionListWrapper.find('dt').exists()).toBe(false);
+  });
+
+  it('Should render workspace list', () => {
+    const { pipeline } = pipelineTestData[PipelineExampleNames.WORKSPACE_PIPELINE];
+    WorkspaceDefinitionListWrapperProps = {
+      workspaces: pipeline.spec.workspaces,
+    };
+    WorkspaceDefinitionListWrapper = shallow(
+      <WorkspaceDefinitionList {...WorkspaceDefinitionListWrapperProps} />,
+    );
+    expect(WorkspaceDefinitionListWrapper.find('dd').exists()).toBe(true);
+    expect(WorkspaceDefinitionListWrapper.find('dt').exists()).toBe(true);
```

Comment:
Testing static markup here makes the test very fragile to component changes. You can omit these tag present assertions.

Reply:
updated